docs/interpreter/r.md

```diff
@@ -0,0 +1,178 @@
+
+[![sparkrfaithful](screenshots/sparkrfaithful.png)](screenshots/sparkrfaithful.png)
+ 
+The Spark Context and SQL Context are created and injected into the local environment automatically as `sc` and `sql`.
+
+The same context are shared with the `%spark`, `%sql` and `%pspark` interpreters:
+
+[![backtoscala](screenshots/backtoscala.png)](screenshots/backtoscala.png)
+
+You can also make an ordinary R variable accessible in scala and Python:
+
+[![varr1](screenshots/varr1.png)](screenshots/varr1.png)
+
+And vice versa:
+
+[![varscala](screenshots/varscala.png)](screenshots/varscala.png)
+[![varr2](screenshots/varr2.png)](screenshots/varr2.png)
+
+### Caveats & Troubleshooting
+
+* The `knitr` environment is persistent. If you run a chunk from Zeppelin that changes a variable, then run the same chunk again, the variable has already been changed.  Use immutable variables. 
+
+* Using the `%r` interpreter, if you return a data.frame, HTML, or an image, it will dominate the result. So if you execute three commands, and one is `hist()`, all you will see is the histogram, not the results of the other commands. This is a Zeppelin limitation.
+
+* If you return a data.frame (for instance, from calling `head()`) from the `%spark.r` interpreter, it will be parsed by Zeppelin's built-in data visualization system.  
+
+*	`htmlwidgets` does not work.  Neither does `shiny` or `ggvis.` 
+
+* Why `knitr` Instead of `rmarkdown`?  Why no `htmlwidgets`?
+
+A major difference between `knitr` and `rmarkdown` is that `rmarkdown` uses `pandoc`, an application external to R.
+
+`htmlwidgets` produces HTML that references dependencies, javacript and css files, that in turn reference other dependencies. 
+
+In `rmarkdown`, `pandoc` processes these dependencies, pulling-in and incorporating the referenced code into a single, self-contained HTML file. 
+
+To use `pandoc`, though, `rmarkdown` writes everything to disc, then calls `pandoc` as an external process, then reads `pandoc`'s output back from disc.  
+
+`knitr`, on the other hand, can operate entirely (or almost entirely) in RAM. 
+
+Its the difference between getting a result in a second, and getting a result in a minimum of a minute.  
+
+After testing, I felt the additional delay would be intolerable for most users and therefore chose to use `knitr` instead of `rmarkdown`.
+
+I would like to support `htmlwidgets` and have done some work on code to handle dependencies, but it is a non-trivial task.
+
+If there is really substantial interest, it isn't *too* challenging to add an `%rmarkdown` interpreter to the mix.
+
+* Why no `ggvis` or `shiny`?
+
+Supporting `shiny` would require integrating a reverse-proxy into Zeppelin, which is a task. 
+
+* But I really want to use `ggvis` and `shiny`!!!!
+
+Then help me write the necessary code.
+
+* I'm having installation problems with Mac OS X
+
+Are you trying to install on a filesystem that is case-insensitive?  
```

Comment:
Here and below I think we better change the style a bit, to be more descriptive and less emotional. 

It seems like here we do not want to know the answer to this question from the user, don't we? Then it's better remove it, as well as mentioning un-related commercial products\companies below because this is going to be official project documentation about R interpreter.
 
Cases-insensitive filesystem is well-known "feature" of os x, and might worth mentioning in "troubleshooting" section.

It's a good to keep Documentation style, so instead of saying "I try" or "I get", (remember, this is the document that many other users will read) it's better just to just keep the facts like

```
Troubleshooting:
 - on os x remember that fs is cases-insensitive
 - `unable to start device X11` - check that DISPLAY env variable is set
 - etc
```